set_person_type_and_uf_affiliation.R

```diff
@@ -0,0 +1,124 @@
+library(tidyverse)
+library(dotenv)
+library(REDCapR)
+library(lubridate)
+library(sendmailR)
+
+source("R/data_fetching.R")
+source("R/data_writing.R")
+
+# set the timezone
+script_name <- "set_person_type_and_uf_affiliation"
+script_run_time <- with_tz(now(), tzone = Sys.getenv("TIME_ZONE"))
+Sys.getenv("INSTANCE")
+
+read_case_data <- read_redcap_db(
+  redcap_uri = Sys.getenv("URI"),
+  token = Sys.getenv("CASE_TOKEN"),
+  project_title = Sys.getenv("CASE_PROJECT_TITLE"),
+  batch_size = 1000
+)
+
+read_contacts_data <- read_redcap_db(
+  redcap_uri = Sys.getenv("URI"),
+  token = Sys.getenv("CONTACT_TOKEN"),
+  project_title = Sys.getenv("CONTACT_PROJECT_TITLE"),
+  batch_size = 1000
+)
+
+read_results_data <- read_redcap_db(
+  redcap_uri = Sys.getenv("URI"),
+  token = Sys.getenv("RESULT_TOKEN"),
+  project_title = Sys.getenv("RESULT_PROJECT_TITLE"),
+  batch_size = 10000
+)
+
+if (length(read_case_data) > 2 &
+    length(read_contacts_data) > 2 &
+    length(read_results_data) > 2) {
+
+  case_data <- read_case_data$data
+  contacts_data <- read_contacts_data$data
+  results_data <- read_results_data$data
+
+  persontype <- results_data %>%
+    select(record_id, person_type) %>%
+    filter(!is.na(person_type))
+
+  set_case_person_type_and_affiliation <- case_data %>%
+    filter(redcap_event_name == "case_investigation_arm_1" &
+             is.na(redcap_repeat_instance)) %>%
+    select(record_id, redcap_event_name, redcap_repeat_instance,
+           case_person_type, case_ufaffiliation, case_emp_ufid) %>%
+    filter(is.na(case_person_type) |
+             (case_ufaffiliation == 0 & !is.na(case_person_type))) %>%
```

Comment:
```suggestion
    filter(is.na(case_person_type)) %>%
```

You really don't want those other records. They all point at bad data we can't fix here. I fixed them manually and have asked @tlstoffs to make changes that might bring down the data entry errors.